specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/queue.json

```diff
@@ -0,0 +1,570 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2019-06-01",
+    "title": "StorageManagementClient",
+    "description": "The Azure Storage Management API."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/queueServices": {
+      "get": {
+        "tags": [
+          "QueueServiceProperties"
+        ],
+        "operationId": "ListQueueService",
```

Comment:
It seems list queue service under a storage account don't have much value, as there's only one queue service in a storage account, Get API should be enough.

Reply:
So we have a valid use case here: There are services that build browsers using the ARM APIs, so they want to list the instances of "tableServices" or queueServices or even blob/fileServices.  For MSFT there is only ever 1 instance. "default". but the tools still need to be able to call the collection API to get the list of one.

Reply:
Exactly right.

---
In reply to: [418164878](https://github.com/Azure/azure-rest-api-specs/pull/9259#discussion_r418164878) [](ancestors = 418164878)